pancancer-arch-3/src/main/java/info/pancancer/arch3/containerProvisioner/ContainerProvisionerThreads.java

```diff
@@ -304,28 +306,27 @@ public Void call() throws Exception {
                         db.updateJobMessage(status.getJobUuid(), status.getStdout(), status.getStderr());
                     }
 
-                    // now update that DB record to be exited
-                    // this is acutally finishing the VM and not the work
-                    if (status.getState() == StatusState.SUCCESS && Utilities.JOB_MESSAGE_TYPE.equals(status.getType())) {
-                        if (!settings.containsKey(Constants.PROVISION_MAX_RUNNING_CONTAINERS)) {
-                            LOG.info("No max_running_containers specified, skipping provision ");
-                            continue;
-                        }
-                        // this is where it reaps, the job status message also contains the UUID for the VM
-                        db.finishContainer(status.getVmUuid());
-                        synchronized (ContainerProvisionerThreads.class) {
-                            runReaper(settings, status.getIpAddress(), status.getVmUuid());
-                        }
-                        if (endless) {
-                            Thread.sleep(MINUTE_IN_MILLISECONDS);
+                    if (Utilities.JOB_MESSAGE_TYPE.equals(status.getType())) {
+                        // now update that DB record to be exited
+                        // this is acutally finishing the VM and not the work
+                        if ((status.getState() == StatusState.SUCCESS)
```

Comment:
This means that successful workers will still get reaped. Is the "endless" mode for workers ever going to be relevant, or is that feature no longer needed/wanted? If it will never be needed, we should probably remove it from the Worker code.